dvc/repo/index.py

```diff
@@ -113,6 +114,16 @@ def is_stage_inside_path(stage: "Stage") -> bool:
 
         return self.filter(is_stage_inside_path)
 
+    def get_key(self, out: "Output") -> Tuple[str, "DataIndexKey"]:
+        if out.is_in_repo:
+            workspace = "repo"
+            key = self.repo.fs.path.relparts(out.fs_path, self.repo.root_dir)
+        else:
+            workspace = out.fs.protocol
+            no_drive = out.fs.path.flavour.splitdrive(out.fs_path)[1]
+            key = out.fs.path.parts(no_drive)[1:]
+        return workspace, key
```

Comment:
Should this be just `out.key`? Long term every output should now its corresponding data index entry and probably key.